ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp

```diff
@@ -2111,12 +2111,13 @@ OGRErr OGRFlatGeobufLayer::ICreateFeature(OGRFeature *poNewFeature)
     // ogrGeometry->exportToWkt(&wkt);
     // CPLDebugOnly("FlatGeobuf", "poNewFeature as wkt: %s", wkt);
 #endif
-    if (ogrGeometry == nullptr || ogrGeometry->IsEmpty())
+    if (m_bCreateSpatialIndexAtClose && (ogrGeometry == nullptr || ogrGeometry->IsEmpty()))
     {
-        CPLDebugOnly("FlatGeobuf", "Skip writing feature without geometry");
-        return OGRERR_NONE;
+        CPLError(CE_Failure, CPLE_AppDefined,
```

Comment:
perhaps we could add some words about that in doc/source/drivers/vector/flatgeobuf.rst in a "Creation issues" paragraph ?